internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -184,6 +194,13 @@ func (u *Upgrader) Upgrade(ctx context.Context, a Action, reexecNow bool) (_ ree
 		return nil, nil
 	}
 
+	// Clean everything from the downloads dir
+	// If we wanted to be a bit simpler we could call os.RemoveAll to remove the dir + children
```

Comment:
That sounds like a plan :)

Reply:
done, also added the call before the exec now call